contrib/babelfishpg_common/src/varbinary.c

```diff
@@ -190,6 +203,12 @@ varbinaryin(PG_FUNCTION_ARGS)
 		PG_RETURN_BYTEA_P(result);
 	}
 
+	if ((!(*pltsql_protocol_plugin_ptr && (*pltsql_protocol_plugin_ptr)->is_tds_client)) && 
```

Comment:
Your condition will be true if pltsql_protocol_plugin_ptr will be NULL as well which is wrong behaviour

Reply:
This is simply a copy of the `IS_TDS_CLIENT()` macro defined in `pltsql.h`:
```
#define IS_TDS_CLIENT() (*pltsql_protocol_plugin_ptr && \
						 (*pltsql_protocol_plugin_ptr)->is_tds_client)
```
I don't see why it is wrong to have the condition be true if the pointer is NULL - the point is that if we are not in a TDS connection, we want to use the default BYTEA conversion.

Reply:
We are defeating the purpose of varbinary datatype altogether, check my other comment where I have explained this in detail

Reply:
Updated to use the existing GUC `psql_logical_babelfish_db_name` - this is a DMS-specific GUC that is not publicly documented anywhere, so it is extremely unlikely that anyone would accidentally set this GUC (unless they were inspecting the open source code). Chatted with Surendra and Bill about this, and they both agree that this is a sufficient restriction on the behavior override.